proto-public/pbmesh/v2beta1/mesh_gateway.proto

```diff
@@ -15,6 +16,9 @@ message MeshGateway {
 
   // +kubebuilder:validation:MinItems=1
   repeated MeshGatewayListener listeners = 2;
+
+  // Selection of workloads to be exposed through the MeshGateway
+  hashicorp.consul.catalog.v2beta1.WorkloadSelector workloads = 3;
```

Comment:
Hmm based on Nathan's comment [here](https://github.com/hashicorp/consul-k8s/pull/3465#pullrequestreview-1815811800), I think you might need to break this out so it doesn't show up on the CRD. I couldn't find a kubetag that ignores a field completely, but it does look like you can set it to [ignore unexported fields ](https://book.kubebuilder.io/reference/controller-gen)

Reply:
Does that mean we would need to add the ignore option on the controller-gen command itself?

Reply:
I also couldn't find an option to ignore a single field; however, we paired as a group earlier today and worked through omitting the field in the consul-k8s CRD. We'll just use a different parent struct for the CRD in consul-k8s and then map that to a `MeshGateway` defined here by adding a workload selector before syncing into consul